console/v1/types_console_plugin.go

```diff
@@ -0,0 +1,197 @@
+	Service ConsolePluginService `json:"service"`
+	// proxy is a list of proxies that describe various service type
+	// to which the plugin needs to connect to.
+	// +kubebuilder:validation:Optional
+	Proxy []ConsolePluginProxy `json:"proxy,omitempty"`
+	// i18n is the configuration of plugin's localization resources.
+	// +kubebuilder:validation:Required
+	I18n ConsolePluginI18n `json:"i18n"`
+}
+
+// LoadType is an enumeration of i18n loading types
+// +kubebuilder:validation:Pattern=`^(Preload|Lazy)$`
+type LoadType string
+
+const (
+	// Preload will load all plugin's localization resources during
+	// loading of the plugin.
+	Preload LoadType = "Preload"
+	// Lazy wont preload any plugin's localization resources, instead
+	// will leave thier loading to runtime's lazy-loading.
+	Lazy LoadType = "Lazy"
+)
+
+// ConsolePluginI18n holds information on localization resources that are served by
+// the dynamic plugin.
+type ConsolePluginI18n struct {
+	// load indicates how the plugin's localization resource should be loaded.
+	// +kubebuilder:validation:Required
+	Load LoadType `json:"load"`
+}
+
+// ConsolePluginProxy holds information on various service types
+// to which console's backend will proxy the plugin's requests.
+type ConsolePluginProxy struct {
+	// backend provides information about endpoint to which the request is proxied to.
+	Endpoint ConsolePluginProxyEndpoint `json:"endpoint"`
+	// alias is a proxy name that identifies the plugin's proxy. An alias name
+	// should be unique per plugin. The console backend exposes following
+	// proxy endpoint:
+	//
+	// /api/proxy/plugin/<plugin-name>/<proxy-alias>/<request-path>?<optional-query-parameters>
+	//
+	// Request example path:
+	//
+	// /api/proxy/plugin/acm/search/pods?namespace=openshift-apiserver
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=128
+	// +kubebuilder:validation:Pattern=`^[A-Za-z0-9-_]+$`
+	Alias string `json:"alias"`
+	// caCertificate provides the cert authority certificate contents,
+	// in case the proxied Service is using custom service CA.
+	// By default, the service CA bundle provided by the service-ca operator is used.
+	// +kubebuilder:validation:Pattern=`^-----BEGIN CERTIFICATE-----([\s\S]*)-----END CERTIFICATE-----\s?$`
+	// +kubebuilder:validation:Optional
+	CACertificate string `json:"caCertificate,omitempty"`
+	// authorization provides information about authorization type,
+	// which the proxied request should contain
+	// +kubebuilder:validation:Optional
+	Authorization AuthorizationType `json:"authorization,omitempty"`
+}
+
+// ConsolePluginProxyEndpoint holds information about the endpoint to which
+// request will be proxied to.
+type ConsolePluginProxyEndpoint struct {
+	// type is the type of the console plugin's proxy. Currently only "Service" is supported.
+	// +kubebuilder:validation:Required
+	Type ConsolePluginProxyType `json:"type"`
```

Comment:
This is the discriminator
```suggestion
	// +unionDisrciminator
	Type ConsolePluginProxyType `json:"type"`
```